anatomy-of-a-social-network.md

```diff
@@ -0,0 +1,180 @@
+for each application to innovate on top of the common protocols.
+Standardise where we see value in standers and compete everywhere else.
+ 
+# Layers
+ 
+## Durability
+There's not much to say here other then the fact that a social network is
+based on the sharing of content so that content must be stored somewhere.
+Whether that is a rotating tape, disk, flash, or nvme is not particularly
+interesting. Even whether it is in an object store, filesystem, or database is
+not that interesting.
+The main thing we need to define in the durability layer is the pattern for 
+maintaining the data.
+Some durability protocols will try to implement permanent storage that always
+repairs faster than it degrades.
+Others will be opportunistic caching that try to be helpful
+but may delete anything at any time.
+Some will be First in first out. Some maintaining for a
+fixed time or a fixed amount of space.
+Some will try to save popular content.
+Some will save content on the verge of extinction.
+It is these policies that will define the durability layer
+not the method of persistence.
+ 
+## The document web
+Social networks are made of a set of artifacts. 
+While the value in the network is in the
+connections between users of the network, 
+the network itself is the connections between documents.
+A post is a document created by a user 
+but that is not the only thing that is a document.
+A comment on that post is itself a document that refers to the original post.
+Even things like reactions which we may not think of as a document
+can be viewed that way.
+"User 1 has liked post 42"
+"User 2 has 😂 ed post 42"
+A post having millions of likes is just an aggregation over the individual likes. 
+While for efficiency we would want materialised views and indexes over 
+the documents these documents
+are the ground truth of a Social network and if we wish to have a 
+Social internetwork we are going to need networks to understand each other's
+documents at least up to a point.
+Profile pages are documents.
+List of users are documents.
+polls and votes are documents.
+Calendar events are documents.
+RSVPs are documents.
+In a public legger system even a payment 
+or asset transfer is just a type of document.
+ 
+The goal of designing a good protocol at the document layer is deciding:
+ What should be defined in the protocol?
+ What should be defined in the standard library of schemas?
+ What should be defined in a public repository of schemas? 
+ What should be defined in local scopes to the applications themselves?
+ 
+Even once we have a solid protocol at this level the schema repositories will
+never be done, always evolving to the needs of current applications.
+ 
+## The content channels
+The documents may be the truth of the Social network 
+but without a way to discover relevant documents are not useful.
+There are two main classes of content
+Channels authoritative and non-authoritative
+ 
+### Authoritative
```

Comment:
I would put this as assertions about a document.  the authoritative nature of them may be a spectrum

Reply:
Added a note to explain that Authoritative just means Author is known not that the post is true or trustworthy.